lib/sql.js

```diff
@@ -1065,6 +1069,10 @@ SQLConnector.prototype.all = function find(model, filter, options, cb) {
       return self.fromRow(model, obj);
     });
     if (filter && filter.include) {
+      if(filter.where) {
```

Comment:
You're right, the place to trigger PARTITION BY injection is incorrect. I removed setting this  option from find() in sql.js and added it to juggler's includeHasMany() in PR https://github.com/strongloop/loopback-datasource-juggler/pull/764 and fixed the 'where' argument passed to buildPartitionBy() in https://github.com/strongloop/loopback-connector-mssql/pull/62. 

The only change left in this PR is the error logging.